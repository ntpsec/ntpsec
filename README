= The NTPsec distribution base directory =

This directory and its subdirectories contain NTPSec, a
security-hardened implementation of Network Time Protocol Version 4.

The contents of the base directory are given in this file. The contents of
subdirectories are usually given in the README files in each subdirectory.

A complete explanation of the configure, compile and install process, as
well as setting up an NTP subnet, is in the HTML pages in the ./html/
directory.

The base directory ./ contains the configuration files, source
directories and related stuff:

INSTALL::	Generic installation instructions.

NEWS::		What's new in this release.

README::	This file.

VERSION::	The version stamp, to be used by scripts and programs.

attic/:: 	Directory containing source code that is *not* part of a
		normal installation. Things can disappear from here at any
		time.

contrib/::	Directory containing contributed scripts, dragons dwell here.
		Some of this might eventually move to being supported code.

devel/::	Documentation and small tools aimed at developers.
		Includes a hacking guide and a tour of the internals.

docs/::		Directory containing a complete set of documentation on
		building and configuring a NTP server or client. The files
		are in asciidoc markup.  This replaces the 'html' directory
		of previous versions, but html can be generated from it.

etc/::		Directory containing a motley collection of configuration files
		and launch scripts for various systems. For example
		only.

include/::	Directory containing include header files used by most
		programs in the distribution.

libisc/::	Utility code from Internet Systems Consortium.  The local
		copy has been significantly trimmed and modified.

libjsmn/::	A minimal JSON library used by the GPSD-JSON driver.

libntp/::	Directory containing library source code used by most
		programs in the distribution.

libparse/:: 	This directory contains the files making up the parser for
		the parse refclock driver. For reasonably sane clocks
		this refclock drivers allows a refclock implementation
		by just providing a conversion routine and the
		appropriate NTP parameters

ntpclients/::   Directory containing sources for clients - utility programs
		to query local and remote NTP installations for log status,
		state variables, and other timekeeping information.  The term
		"clients" is used quite loosely here; any tool that is not a
		not a multi-file C program probably lives in this directory.

ntpd/::		Sources for the main time-synchronization daemon.

ntpfrob/::      The ntpfrob utility collects several small diagnostic
		functions for reading and tweaking the local clock
		hardware, including reading the clock tick rate,
		precision, and jitter.

ntptime/:: 	Directory containing a utility for reading and modifying
		kernel parameters related to the local clock.

pylib/::	Installable Python helper classes for scripts.

tests/::	Self-test code.

waf::		A copy of the waf builder.  This is the engine used to configure
		and build the codebase.

wafhelpers/::	A library of Python procedures used by the waf build system.

wscript::	NTP-specific waf rules.

// end
