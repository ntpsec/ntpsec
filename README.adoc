= About this fork =

This fork's only reason is to create a link:https://github.com/ntpsec/ntpsec/pull/8[pull request].
Only the ntpdig client is affected.

= The NTPsec distribution base directory =

This directory and its subdirectories contain NTPSec, a
security-hardened implementation of Network Time Protocol Version 4.
You can browse a summary of differences from legacy NTP here:

https://docs.ntpsec.org/latest/ntpsec.html

The contents of the base directory are given in this file. The contents of
subdirectories are usually given in the README files in each subdirectory.

The base directory ./ contains the configuration files, source
directories and related stuff:

INSTALL.adoc::	Generic installation instructions.

NEWS.adoc::	What's new in this release.

README.adoc::	This file.

LICENSES/::	License texts in REUSE format. See http://reuse.software

VERSION::	The version stamp, to be used by scripts and programs.

attic/::	Directory containing source code that is *not* part of a
		normal installation. Things can disappear from here at any
		time.

buildprep::	Executable script for fetching installation prerequisites.

contrib/::	Directory containing contributed scripts, dragons dwell here.
		Some of this might eventually move to being supported code.

devel/::	Documentation and small tools aimed at developers.
		Includes a hacking guide and a tour of the internals.

docs/::		Directory containing a complete set of documentation on
		building and configuring a NTP server or client. The files
		are in asciidoc markup.  This replaces the 'html' directory
		of previous versions, but html can be generated from it.

etc/::		Directory containing a motley collection of configuration files
		and launch scripts for various systems. For example
		only.

include/::	Directory containing include header files used by most
		programs in the distribution.

libjsmn/::	A minimal JSON library used by the GPSD-JSON driver.

libntp/::	Directory containing library source code used by most
		programs in the distribution.

libparse/::	This directory contains the files making up the parser for
		the parse refclock driver. For reasonably sane clocks
		this refclock drivers allows a refclock implementation
		by just providing a conversion routine and the
		appropriate NTP parameters

ntpclients/::   Directory containing sources for clients - utility programs
		to query local and remote NTP installations for log status,
		state variables, and other timekeeping information.  The term
		"clients" is used quite loosely here; any tool that is not a
		multi-file C program probably lives in this directory.

ntpd/::		Sources for the main time-synchronization daemon.

ntpfrob/::      The ntpfrob utility collects several small diagnostic
		functions for reading and tweaking the local clock
		hardware, including reading the clock tick rate,
		precision, and jitter.

ntptime/::	Directory containing a utility for reading and modifying
		kernel parameters related to the local clock.

packaging/::	Parts and guidance for distribution packagers.

pylib/::	Installable Python helper modules for scripts.

tests/::	Self-test code.

waf::		A copy of the waf builder.  This is the engine used to configure
		and build the codebase.

wafhelpers/::	A library of Python procedures used by the waf build system.

wscript::	NTP-specific waf rules.

www/::		Sample ntpviz files

// end
