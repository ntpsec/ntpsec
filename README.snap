= Using NTPsec from a snap =

Snaps provide further isolation and an alternative cross-distribution mechanism
to provide applications.

Due to that isolation the tools you know are exposed to the host (your normal
system) via a prefix. For example ntpq becomes ntpseq.ntpq.

Currently the following binaries are exposed by the ntpsec snap:
 - ntpsec.ntpdig
 - ntpsec.ntpfrob
 - ntpsec.ntpkeygen
 - ntpsec.ntpq
 - ntpsec.ntptime

Additionally ntpd is exposed as a daemon (via a systemd service).
It can be accessed via the usual tools as snap.ntpsec.ntpd.service.

Examples:
 1. get status / start / stop / restart
   $ systemctl status snap.ntpsec.ntpd.service

 2. check the log just for this service
   $ journalctl -u snap.ntpsec.ntpd

// end
